Release Notes 1.0 (#130)
percona/everest-doc: docs/release-notes/Percona-Everest-1.0.0-(2024-06-28).md

# What's new in Percona Everest 1.0.0

We proudly announce that Percona Everest has officially hit the general availability (GA) milestone with the release of version 1.0.0.

To begin your journey with Percona Everest, check out the [Quickstart Guide for Percona Everest](../quickstart-guide/quick-install.md).

Percona Everest is an open source cloud native database platform that helps provision and manage databases faster, scale deployments rapidly, and reduce database administration overhead. Plus, you can regain control over your data, database configuration, and DBaaS costs.

Version 1.0.0 introduces the following changes:

## Release highlights

## Simplified database operator upgrades

We are excited to announce that you can now upgrade database operators and all their components across any namespace with just a single click using our intuitive UI.

> ![!image](../images/upgrade_operator.png)

Author: They will be visible after this branch is merged in the Release branch.

To explore the depths of this feature, delve into our [documentation](https://docs.percona.com/everest/secure/IdP_integratiopn.html).

## New features

- [EVEREST-816](https://perconadev.atlassian.net/browse/EVEREST-816) - Starting with Percona Everest 1.0.0, you can now upgrade database operators and all their components across any namespace with just a single click using our intuitive UI.

- [EVEREST-1087](https://perconadev.atlassian.net/browse/EVEREST-1087) - You can now integrate your Percona Everest instance using an external identity provider (IdP). This enables centralized authentication and authorization management, streamlining and simplifying user access.

- [EVEREST-1025](https://perconadev.atlassian.net/browse/EVEREST-1025) - We introduced the user management feature with Percona Everest 1.0.0, enabling you to securely log in to the platform through either the user interface or the API.

> - [EVEREST-731](https://perconadev.atlassian.net/browse/EVEREST-731) Add support for new Load balancer setting

Reviewer: ```suggestion
- [EVEREST-731](https://perconadev.atlassian.net/browse/EVEREST-731) Add support for customizing load balancer source ranges in PostgreSQL clusters
```

> - [EVEREST-866](https://perconadev.atlassian.net/browse/EVEREST-866) - In the past, when viewing the DB cluster in the backups list, it was confusing to find information about the Point-in-Time Recovery (PITR) for restores. With the release of Percona Everest 1.0.0, PITR is now visible. However, PITR is only available to select if the chosen backup is the most recent full backup.

Reviewer: I believe you got this description wrong. This ticket is about having the PITR option available only if the last backup is selected.

Author: fixed

- [EVEREST-872](https://perconadev.atlassian.net/browse/EVEREST-872) - When deleting a backup, you can now choose to delete the data from the backup storage as well.

> - [EVEREST-1095](https://perconadev.atlassian.net/browse/EVEREST-1095) - The new API method GET /v1/settings has been added to the Everest API. This method does not require the auth token so that the UI can get the initial Everest settings.
> 
> 

Reviewer: I think we should remove this point. Right now this is only useful for the IdP integration which we already have a bullet so it serves no purpose
```suggestion
```

## Improvements

> - [EVEREST-937](https://perconadev.atlassian.net/browse/EVEREST-937) - We have made some improvements in our telemetry, including sending telemetry data about the DB cluster every time a user creates one and adding information about the Everest version reported for the instance ID.

Reviewer: I'm not sure we should even have this in the RN because this isn't a feature that our users care about, it's just for our benefit. If we decide to keep it let's at least move this to the bottom.

Author: @recharte Peter asked me to keep this one. So, let's keep it, but as you suggested, let's move it to the bottom.

- [EVEREST-924](https://perconadev.atlassian.net/browse/EVEREST-924) - Starting with Percona Everest 1.0.0, you now have the option to create multiple backup schedules using the wizard.

- [EVEREST-931](https://perconadev.atlassian.net/browse/EVEREST-931) - When you go through a wizard, return to a specific step, and delete something from a mandatory field, the editing functionality is now disabled.

- [EVEREST-953](https://perconadev.atlassian.net/browse/EVEREST-953) - For an improved user interface (UI) experience, we have consolidated backups and PITR on the same page.

- [EVEREST-971](https://perconadev.atlassian.net/browse/EVEREST-971) - Access and secret key inputs are now visible on the UI when adding a storage location. You can use the eye icon to toggle between making the keys visible or hidden. This feature allows you to conveniently view the S3 keys directly from the UI.

- [EVEREST-974](https://perconadev.atlassian.net/browse/EVEREST-974) - Everest now supports editing the DB Engine version after a cluster has been created. However, it's important to note the following restrictions:

    - You are unable to upgrade to a different major version. 

>     - Downgrading the DB Engine version is not supported.

Reviewer: Maybe this should be part of the new features

Author: Done

- [EVEREST-1007](https://perconadev.atlassian.net/browse/EVEREST-1007) - For an improved user experience, the **Actions** button has been moved to the **Database Details** tab on the right side of the database name.

- [EVEREST-1055](https://perconadev.atlassian.net/browse/EVEREST-1055) Starting with Percona Everest 1.0.0, we have introduced a new **deleting** state. This state will persist until all resources associated with the database have been removed.

- [EVEREST-1069](https://perconadev.atlassian.net/browse/EVEREST-1069) We are constantly working to enhance our user experience, and we've recently introduced a new page - the components page. This page provides detailed information about the pods and containers, including their status, type, age, and more.

> 

Reviewer: This is definitely a new feature and I think it deserves its spot in the release highlights :stuck_out_tongue:

Author: Done

> - [EVEREST-1093](https://perconadev.atlassian.net/browse/EVEREST-1093) - We now offer users the choice to select or deselect **Delete backup storage data** when attempting to delete a database without backups.

Reviewer: This description is wrong and I don't think we need to advertise this at all.
Instead, we should add to the new features section this ticket:
https://perconadev.atlassian.net/browse/EVEREST-873

Author: Done.

### Bugs

- [EVEREST-807](https://perconadev.atlassian.net/browse/EVEREST-807) - Fixed an issue where PITR did not display the storage location being used when enabling PITR during database creation or editing.

- [EVEREST-837](https://perconadev.atlassian.net/browse/EVEREST-837) - We have now updated the help for the Command Line Interface (CLI) commands to include the descriptions.

- [EVEREST-841](https://perconadev.atlassian.net/browse/EVEREST-841) - Fixed an issue where the user interface (UI) could not identify the correct operator/database cluster for different namespaces.

> - [EVEREST-859](https://perconadev.atlassian.net/browse/EVEREST-859) - While uninstalling Percona Everest, an error occurred when attempting to delete a namespace. The issue is resolved now.
> 

Reviewer: Apparently, we are still facing this issue, let's remove it from this list.
```suggestion
```

- [EVEREST-870](https://perconadev.atlassian.net/browse/EVEREST-870) - When running the `everestctl install` command, the installation wizard asked for values such as namespaces and operators, even though the values were already provided by flags `(--namespaces=everest --operator.mongodb=false --operator.postgresql=false --operator.xtradb-cluster=true)`. The issue has been resolved now.

- [EVEREST-1003](https://perconadev.atlassian.net/browse/EVEREST-1003) - Resolved an issue where the installation of operators in a new namespace was failing.

- [EVEREST-1016](https://perconadev.atlassian.net/browse/EVEREST-1016) - We updated the **Last backup** status from **inactive** to **scheduled** because it was confusing for the users.

- [EVEREST-1034](https://perconadev.atlassian.net/browse/EVEREST-1034) - The Restores page did not display the restores in a sorted order. The issue has been resolved now.

- [EVEREST-1050](https://perconadev.atlassian.net/browse/EVEREST-1050) - The information about the restores was not correctly updated on the restores page. The issue has been fixed now.

> - [EVEREST-1137](https://perconadev.atlassian.net/browse/EVEREST-1137) - When external access was enabled, the external IP address for MySQL was not displayed. The issue has been resolved now.
> 
> 

Reviewer: We found this issue during our development testing and it  wasn't in the previous release so it never affected our users, let's remove it.
```suggestion
```

> - [EVEREST-1139](https://perconadev.atlassian.net/browse/EVEREST-1139) - Addressed an issue where the PostgreSQL database became unresponsive and stuck when attempting to delete the database.
> 

Reviewer: We found this issue during our development testing and it wasn't in the previous release so it never affected our users, let's remove it.
```suggestion
```

- [EVEREST-1087](https://perconadev.atlassian.net/browse/EVEREST-1087) - You can now integrate your Percona Everest instance using an external identity provider (IdP). This enables centralized authentication and authorization management, streamlining and simplifying user access.

- [EVEREST-1025](https://perconadev.atlassian.net/browse/EVEREST-1025) - We introduced the user management feature with Percona Everest 1.0.0, enabling you to securely log in to the platform through either the user interface or the API.

- [EVEREST-974](https://perconadev.atlassian.net/browse/EVEREST-974) - Everest now supports editing the DB Engine version after a cluster has been created. However, it's important to note the following restrictions:

    - You are unable to upgrade to a different major version. 
    - Downgrading the DB Engine version is not supported.

- [EVEREST-1069](https://perconadev.atlassian.net/browse/EVEREST-1069) - We've recently introduced a new page - the components page. This page provides detailed information about the pods and containers, including their status, type, age, and more.

> - [EVEREST-866](https://perconadev.atlassian.net/browse/EVEREST-866) - Starting with Percona Everest 1.0.0, you can restore your database from a full back or using the PITR. However, if you choose a backup other than the latest backup, the PITR option becomes unavailable.

Reviewer: ```suggestion
- [EVEREST-866](https://perconadev.atlassian.net/browse/EVEREST-866) - Starting with Percona Everest 1.0.0, you can restore your database from a full backup or using the PITR. However, if you choose a backup other than the latest backup, the PITR option becomes unavailable.
```
